ArduCopter/AP_Arming.cpp

```diff
@@ -531,12 +533,13 @@ bool AP_Arming_Copter::mandatory_gps_checks(bool display_failure)
         if (reason == nullptr || strlen(reason) == 0) {
             if (!mode_requires_gps && fence_requires_gps) {
                 // clarify to user why they need GPS in non-GPS flight mode
-                reason = "Fence enabled, need 3D Fix";
+                check_failed(display_failure, "Fence enabled, need GPS Fix %u", gps.status_arm_min());
             } else {
-                reason = "Need 3D Fix";
+                check_failed(display_failure, "Needs GPS Fix %u", gps.status_arm_min());
             }
+        } else {
+            check_failed(display_failure, "%s", reason);
```

Comment:
Are these changes needed? I think these still rely on 3D fix, not whatever is configured in gps.status_arm_min(), right?

Reply:
Given that we are stopping arming based on the minimum fix needed based on the param I would say we should display the min fix needed rather than 3D, which could be confusing based on what the param value is set to.

Reply:
Done